### PR TITLE
display: add frame_incomplete

### LINKED ISSUE #79798: display: Support page flipping
**Is your enhancement proposal related to a problem? Please describe.**
We have a display that supports writing to it and then page flipping to update all pixels on the screen at once.

Sadly, Zephyr's display API only supports this partially:
- `SCREEN_INFO_DOUBLE_BUFFER` already exists (for probably this very reason)
- No `pageflip` api function exists yet

Further, LVGL supports this (`lv_disp_flush_is_last`), but it does not get used by Zephyr yet.

**Describe the solution you'd like**
- Implement a `display_pageflip` Display Subsystem API function
- Use `lv_disp_flush_is_last` and call the `display_pageflip` function to pass this information on to the display

**Describe alternatives you've considered**
Instead of adding a `display_pageflip`, this could be an extra flag in the `display_write` function. Although this would be a breaking change.

### PR DESCRIPTION
**Pull request by Finomnis**
Rework of #79936.
Fixes #79798.

Adds the new flag `frame_incomplete` to `struct display_buffer_descriptor` that indicates that more writes will follow in the current frame.

This allows GUIs to communicate where, at multi-write frames, the frame ends.
This in turn allows display to understand the frame border, which can be useful for page flipping or "tearing enable" signal handling.

For more discussion, see #79936.

The effect of this change is that displays can now present multi-writes at once, to prevent the 'rolling shutter' effect of multiple separate partial writes, like so:

Before:
without_view

After:
with_view

### CODE REVIEW

File: drivers/display/display_ili9xxx.c [hunk 1]
```diff
@@ -163,6 +163,7 @@ static int ili9xxx_write(const struct device *dev, const uint16_t x,
 	mipi_desc.width = desc->width;
 	/* Per MIPI API, pitch must always match width */
 	mipi_desc.pitch = desc->width;
+	mipi_desc.frame_incomplete = desc->frame_incomplete;
```
Comment (danieldegrasse) [inline drivers/display/display_ili9xxx.c:166]: Just FYI- we can also add this to the ST7796S display driver

File: drivers/display/display_sdl_bottom.c [hunk 2]
```diff
@@ -126,17 +127,37 @@ int sdl_display_read_bottom(const uint16_t height, const uint16_t width,
 	return err;
 }
 
-void sdl_display_blanking_off_bottom(void *renderer, void *texture)
+void sdl_display_blanking_off_bottom(void *renderer, void *texture, void *mutex)
 {
+	int err;
+
+	err = SDL_TryLockMutex(mutex);
```
Comment (danieldegrasse) [inline drivers/display/display_sdl_bottom.c:134]: Assuming the mutex lock was just absent here before, and this is fixing that? We should split the into another commit if so. Is there a specific reason we need to lock the mutex here? Or is it just to improve the thread safety of this driver?

Comment (Finomnis) [reply drivers/display/display_sdl_bottom.c:134]: Tbh this entire thing is not thread safe at all, including the sdl calls, so I should probably revert this

File: drivers/display/display_ili9xxx.c (hunk 1 above)
Comment (Finomnis) [reply drivers/display/display_ili9xxx.c:166]: I'm not sure what exactly to change here, I just tried to fix it to my best knowledge. Feel free to tell me exactly where to change things

Comment (danieldegrasse) [reply drivers/display/display_ili9xxx.c:166]: This is the right place :) I was just mentioning that other drivers like the ST7796 can use this as well. It is definitely a nonblocking suggestion, this can always be added after the fact. One thing we probably should do though is zero-initialize the `mipi_desc` structures in these drivers in places they aren't already.

Comment (Finomnis) [reply drivers/display/display_ili9xxx.c:166]: Yah, that's why this change is there. I didn't initially find more, so if you have more, feel free to tell me.
The others I found are already zero initialized as previously discussed.

File: drivers/display/display_sdl_bottom.c (hunk 2 above)
Comment (Finomnis) [reply drivers/display/display_sdl_bottom.c]: Fixed

File: samples/drivers/display/src/main.c [hunk 3]
```diff
@@ -296,6 +296,7 @@ int main(void)
 	buf_desc.pitch = capabilities.x_resolution;
 	buf_desc.width = capabilities.x_resolution;
 	buf_desc.height = h_step;
+	buf_desc.frame_incomplete = false;
```
Comment (danieldegrasse) [inline samples/drivers/display/src/main.c:299]: We should be setting `frame_incomplete` to true when we write the last buffer before we call `display_blanking_off`, right? I'm just thinking this would be useful to demonstrate the usage of this field

Comment (Finomnis) [reply samples/drivers/display/src/main.c:299]: I'm unsure what you mean, isn't that frame already complete?

Comment (Finomnis) [reply samples/drivers/display/src/main.c:299]: Sorry, my bad! Didn't realize there are multiple writes. Yes, weil change that.

File: drivers/display/display_ili9xxx.c (hunk 1 above)
Comment (Finomnis) [reply drivers/display/display_ili9xxx.c:166]: @danieldegrasse Fixed

File: samples/drivers/display/src/main.c (hunk 3 above)
Comment (Finomnis) [reply samples/drivers/display/src/main.c]: Fixed

**Review by danieldegrasse — APPROVED** [review]

**Review by faxe1008 — APPROVED** [review]

**Review by kartben — APPROVED** [review]
